Jenkinsfile

```diff
@@ -59,7 +63,11 @@ pipeline {
                         env.testProvider = "nfs"
                     } else if (body.contains("[test yugabytedb]")) {
                         env.testProvider = "yugabytedb"
+                    } else if (!changed_files.contains(".go")) {
```

Comment:
Only checking for .go files makes me nervous. How about if we also trigger the build if we find anything in the `build`, `cmd`, `images`, `pkg`, or `tests` folders?

Reply:
Automating this at all makes me a little nervous. Is it not okay just to use `[skip ci]`?

Reply:
We have certainly been using [skip ci] for this issue for a long time, but it is really quite tedious since there are many PRs that don't need to build. **Rarely** does anyone read the checklist before opening the PR, unfortunately. If we can reliably detect the majority of cases where a build isn't needed it will be a big help to reduce CI resource consumption and maintainers who keep an eye on the CI.

Reply:
This is getting complicated, what about checking for `.go` but have `[force ci]` along with that so we can override? I think this will work most of the time.

How about the new approach?